pytorch_lightning/profiler/profilers.py

```diff
@@ -287,9 +296,47 @@ def __del__(self):
             self.output_file.close()
 
 
-class PyTorchProfiler(BaseProfiler):
+class RegisterRecordFunction:
+
+    def __init__(self, model):
+        self._model = model
+        self._records = {}
+
+    def _start_recording(self, module, input, module_name: str = None, is_built_in: bool = None):
+        if module_name is not None:
+            record_name = module_name if is_built_in else f"{type(module)}: {module_name}"
+            self._records[record_name] = record_function(record_name).__enter__()
+        return input
+
+    def _stop_recording(self, module, input, result, module_name: str = None, is_built_in: bool = None):
+        if module_name is not None:
+            record_name = module_name if is_built_in else f"{type(module)}: {module_name}"
+            self._records[record_name].__exit__(None, None, None)
+        return result
+
+    def __enter__(self):
+        built_in_modules = dir(torch.nn)
+        for module_name, module in self._model.named_modules():
+            is_built_in = module in built_in_modules
+            module.register_forward_pre_hook(
+                partial(self._start_recording, module_name=module_name, is_built_in=is_built_in)
+            )
+            module.register_forward_hook(
+                partial(self._stop_recording, module_name=module_name, is_built_in=is_built_in)
+            )
+
+    def __exit__(self, exc_type: Any, exc_value: Any, traceback: Any):
+        for module in self._model.modules():
+            module._forward_pre_hooks = []
+            module._forward_hooks = []
```

Comment:
What if there are other hooks defined for other things?

Reply:
Great catch !